### PR TITLE
feat: add fscloud profile and example

### PR DESCRIPTION
**Pull request by maheshwarishikha**
### Description

Adds fscloud profile and example

#### No release required

- [x] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [x] CI-related update (pipeline, etc.)
- [ ] Other changes that don't affect Terraform code

#### Release required

- [x] Bug fix (patch release (`x.x.X`): Change that fixes an issue and is compatible with earlier versions)
- [x] New feature (minor release (`x.X.x`): Change that adds functionality and is compatible with earlier versions)
- [ ] Breaking change (major release (`X.x.x`): Change that is likely incompatible with previous versions)

##### Release notes content

Replace this text with information that users need to know about the bug fixes, features, and breaking changes. This information helps the merger write the commit message that is published in the release notes for the module.

---

### Checklist for reviewers

- [ ] The PR references a GitHub issue.
- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Merge by using "Squash and merge".
- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author.

    The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).

### CODE REVIEW

**Review by daniel-butler-irl — CHANGES_REQUESTED** [review]
A few tweaks are needed. I saw some stuff I also need to go back and fix in my PR

File: examples/fscloud/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# FS Cloud profile example
```
[the remark this answers is not in the document]
Comment (maheshwarishikha) [reply examples/fscloud/README.md]: Made the required changes.

File: examples/fscloud/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
```
[the remark this answers is not in the document]
Comment (maheshwarishikha) [reply examples/fscloud/README.md:2]: Done.

File: examples/fscloud/main.tf [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  name                = "${var.prefix}-postgres"
+  region              = var.region
+  pg_version          = var.pg_version
+  key_protect_key_crn = module.key_protect_all_inclusive.keys["icd-pg.${var.prefix}-pg"].crn
+  resource_tags       = var.resource_tags
+  allowlist           = var.allowlist
+  cbr_rules = [
+    {
+      name             = var.pg_version == null ? "${var.prefix}-postgres-zone" : "${var.prefix}-${var.pg_version}-postgres-zone"
+      description      = "sample rule"
```
[the remark this answers is not in the document]
Comment (maheshwarishikha) [reply examples/fscloud/main.tf]: Modified.

File: examples/fscloud/outputs.tf [hunk 4]
```diff
@@ -0,0 +1,12 @@
+##############################################################################
+# Outputs
+##############################################################################
+output "id" {
+  description = "Postgresql instance id"
+  value       = module.postgresql_db.id
+}
+
```
[the remark this answers is not in the document]
Comment (maheshwarishikha) [reply examples/fscloud/outputs.tf:8]: Added

File: examples/fscloud/provider.tf [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  create_returns_object = false
+  debug                 = false # set to true to show detailed logs, but use carefully as it might print sensitive values.
+  headers = {
+    Authorization    = data.ibm_iam_auth_token.token_data.iam_access_token
+    Bluemix-Instance = module.key_protect_all_inclusive.key_protect_guid
+    Content-Type     = "application/vnd.ibm.kms.policy+json"
+  }
+}
+
+provider "restapi" {
```
[the remark this answers is not in the document]
Comment (maheshwarishikha) [reply examples/fscloud/provider.tf]: done

File: profiles/fscloud/main.tf [hunk 6]
```diff
@@ -0,0 +1,25 @@
+
+locals {
+  # tflint-ignore: terraform_unused_declarations
+  validate_restrictions_set = (var.allowlist == null && var.cbr_rules == null) ? tobool("Allow list and/or CBR Rules must be set") : true
```
[the remark this answers is not in the document]
Comment (maheshwarishikha) [reply profiles/fscloud/main.tf]: Made changes

File: profiles/fscloud/outputs.tf [hunk 7]
```diff
@@ -0,0 +1,13 @@
+##############################################################################
+# Outputs
+##############################################################################
+
+output "id" {
+  description = "Postgresql instance id"
+  value       = module.postgresql_db.id
+}
+
```
[the remark this answers is not in the document]
Comment (maheshwarishikha) [reply profiles/fscloud/outputs.tf:9]: Added

File: profiles/fscloud/variables.tf [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+variable "name" {
+  type        = string
+  description = "Name of the Postgresql instance"
+}
+
+variable "pg_version" {
+  description = "Version of the postgresql instance"
+  type        = string
+  default     = null
+  validation {
```
[the remark this answers is not in the document]
Comment (maheshwarishikha) [reply profiles/fscloud/variables.tf]: Removed validation for all the variables.

File: profiles/fscloud/variables.tf [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+    rule_contexts = list(object({
+      attributes = optional(list(object({
+        name  = string
+        value = string
+    }))) }))
+    enforcement_mode = string
+    tags = optional(list(object({
+      name  = string
+      value = string
+    })))
```
[the remark this answers is not in the document]
Comment (maheshwarishikha) [reply profiles/fscloud/variables.tf:154]: why is it required in `type`, shouldn't be taken care while defining the value?

[the remark this answers is not in the document]
Comment (maheshwarishikha) [reply profiles/fscloud/variables.tf]: ok, understood. Made the changes.

File: examples/fscloud/main.tf [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+  pg_version          = var.pg_version
+  key_protect_key_crn = module.key_protect_all_inclusive.keys["icd-pg.${var.prefix}-pg"].crn
+  resource_tags       = var.resource_tags
+  allowlist           = var.allowlist
+  cbr_rules = [
+    {
+      description      = "${var.prefix}-postgres access only from vpc"
+      enforcement_mode = "enabled" #Postgresql does not support report mode
+      account_id       = data.ibm_iam_account_settings.iam_account_settings.account_id
+      tags = [
```
Comment (daniel-butler-irl) [inline examples/fscloud/main.tf:95]: From yesterdays deep dive we will completely remove tags from CBRs

File: main.tf [hunk 11]
```diff
@@ -116,14 +116,9 @@ module "cbr_rule" {
         operator = "stringEquals"
       }
     ],
-    tags = var.cbr_rules[count.index].tags != null ? var.cbr_rules[count.index].tags : [
-      {
-        name  = "terraform-rule"
-        value = "allow-postgresql"
-      }
-    ]
+    tags = var.cbr_rules[count.index].tags
```
Comment (daniel-butler-irl) [inline main.tf:119]: same as a bove remove tags completely

File: examples/fscloud/main.tf (hunk 10 above)
Comment (maheshwarishikha) [reply examples/fscloud/main.tf:95]: As we are removing tags completely, then we need to remove this from variable type format too....is it right?

Comment (maheshwarishikha) [reply examples/fscloud/main.tf]: Made the changes.

File: main.tf (hunk 11 above)
Comment (maheshwarishikha) [reply main.tf]: done.

**Review by daniel-butler-irl — APPROVED** [review]

**Pull request merged**